connectors/elasticsearch/connector.go

```diff
@@ -0,0 +1,441 @@
+			config: cfg,
+			client: client,
+		}, nil
+	})
+}
+
+var _ connectors.Connector = (*Connector)(nil)
+
+// Connector implements the connectors.Connector interface for Elasticsearch
+type Connector struct {
+	config Config
+	client *elasticsearch.Client
+}
+
+func (c *Connector) Config() connectors.Config {
+	return &c.config
+}
+
+// Ping checks if Elasticsearch is reachable
+func (c *Connector) Ping(ctx context.Context) error {
+	res, err := c.client.Ping(c.client.Ping.WithContext(ctx))
+	if err != nil {
+		return fmt.Errorf("failed to ping Elasticsearch: %w", err)
+	}
+	defer res.Body.Close()
+	return nil
+}
+
+// Query executes a search query in Elasticsearch
+func (c *Connector) Query(ctx context.Context, endpoint model.Endpoint, params map[string]any) ([]map[string]any, error) {
+	processed, err := castx.ParamsE(endpoint, params)
+	if err != nil {
+		return nil, xerrors.Errorf("unable to process params: %w", err)
+	}
+
+	var queryMap map[string]any
+	err = json.Unmarshal([]byte(endpoint.Query), &queryMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse query JSON: %w", err)
+	}
+
+	// Replace placeholders inside query efficiently
+	replacePlaceholders(queryMap, processed)
+
+	queryBytes, err := json.Marshal(queryMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal search query: %w", err)
+	}
+	log.Println(queryMap)
+	// Execute the search request
+	res, err := c.client.Search(
+		c.client.Search.WithContext(ctx),
+		c.client.Search.WithBody(bytes.NewReader(queryBytes)),
+	)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to execute search query: %w", err)
+	}
+
+	defer res.Body.Close()
+
+	// Read response body
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to read Elasticsearch response: %w", err)
+	}
+
+	// Check for errors
+	if res.IsError() {
+		return nil, xerrors.Errorf("Elasticsearch returned an error: %s", body)
+	}
+
+	// Parse JSON response
+	var result map[string]interface{}
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to parse Elasticsearch response: %w", err)
+	}
+
+	var hits []interface{}
+	if hitsMap, ok := result["hits"].(map[string]interface{}); ok {
+		if hitsList, ok := hitsMap["hits"].([]interface{}); ok {
+			hits = hitsList
+		} else {
+			return nil, xerrors.Errorf("'hits' key is missing or not a list")
+		}
+	} else {
+		return nil, xerrors.Errorf("'hits' key is missing or not a map")
+	}
+	// Process the results
+	results := make([]map[string]interface{}, 0)
+	for _, hit := range hits {
+		hitMap, ok := hit.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		source, ok := hitMap["_source"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		results = append(results, source)
+	}
+
+	return results, nil
+}
+
+// replacePlaceholders replaces parameters in the "match" clause and other necessary fields
+func replacePlaceholders(query map[string]any, params map[string]any) {
```

Comment:
this is force to support only strng match parametrized-queries, maybe it's better to support instead of that a templated query?
```
{
					"query": {
						"match": {
							"job": ":${.job}"
						}
					}
				}
```

by utilize `query := elastic.NewTemplateQuery("find_user_tpl")`  func?

Reply:
check this test https://github.com/olivere/elastic/blob/master/search_queries_template_query_test.go#L13